app/controllers/drug_stocks_controller.rb

```diff
@@ -0,0 +1,75 @@
+# This controller is meant to be used from the Android app _only_,
+# hence we handle authentication ourselves from params passed from the client.
+class DrugStocksController < ApplicationController
```

Comment:
To clearly differentiate this from `MyFacilities::DrugStocksController`, should we also nest this under something relating to the android app? `ProgressTab::DrugStocksController`, or `WebView::DrugStocksController`? That would also provide the context of the auth being for a webview in a mobile app?

Reply:
I'm not familiar enough with this project to have an informed opinion, but it seems as though this is intended to be an android web view and that it can't be used by the web due to differences in the auth process. In which case, it would probably be wise to separate out the auth logic into a parent controller like `WebViewController`. That might be a pre-optimization that's not necessary for this ticket, but moving it to another namespace like `WebView` might be a good start.

Reply:
I agree on the namespace 👍 . I also agree that a parent controller is premature, as there is just one controller for this type of approach so far.